common/lib/dependabot/package_manager.rb

```diff
@@ -0,0 +1,52 @@
+# typed: strong
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+
+module Dependabot
+  class PackageManagerBase
+    extend T::Sig
+    extend T::Helpers
+
+    abstract!
+
+    # The name of the package manager (e.g., "bundler")
+    sig { abstract.returns(String) }
+    def name; end
+
+    # The version of the package manager (e.g., "2.1.4")
+    sig { abstract.returns(T.nilable(String)) }
```

Comment:
For consistency too; should this be the `Version` object we already have in dependabot?

Reply:
That way we can simplify and standardize - e.g. there wouldn't be a need for the major_version method